examples/container-service-extension-4.0/install/step2/3.9-cse-4.0-install-step2.tf

```diff
@@ -0,0 +1,779 @@
+    "${data.vcd_rde_type.existing_capvcdcluster_type.vendor}:${data.vcd_rde_type.existing_capvcdcluster_type.nss}: Administrator View",
+    "vmware:tkgcluster: Full Access",
+    "vmware:tkgcluster: Modify",
+    "vmware:tkgcluster: View",
+    "vmware:tkgcluster: Administrator View",
+    "vmware:tkgcluster: Administrator Full access",
+  ]
+
+  publish_to_all_tenants = true # This needs to be published to all the Organizations
+
+  # As we use rights created by the CAPVCD Type created previously, we need to depend on it
+  depends_on = [
+    vcd_rights_bundle.k8s_clusters_rights_bundle
+  ]
+}
+
+# The networking setup specified below will configure one Provider Gateway + Edge Gateway + Routed network per
+# organization. You can customise this section according to your needs.
+
+data "vcd_nsxt_manager" "cse_nsxt_manager" {
+  name = var.nsxt_manager_name
+}
+
+data "vcd_nsxt_tier0_router" "solutions_tier0_router" {
+  name            = var.solutions_nsxt_tier0_router_name
+  nsxt_manager_id = data.vcd_nsxt_manager.cse_nsxt_manager.id
+}
+
+resource "vcd_external_network_v2" "solutions_tier0" {
+  name = "solutions_tier0"
+
+  nsxt_network {
+    nsxt_manager_id      = data.vcd_nsxt_manager.cse_nsxt_manager.id
+    nsxt_tier0_router_id = data.vcd_nsxt_tier0_router.solutions_tier0_router.id
+  }
+
+  ip_scope {
+    gateway       = var.solutions_provider_gateway_gateway_ip
+    prefix_length = var.solutions_provider_gateway_gateway_prefix_length
+
+    dynamic "static_ip_pool" {
+      for_each = var.solutions_provider_gateway_static_ip_ranges
+      iterator = ip
+      content {
+        start_address = ip.value[0]
+        end_address   = ip.value[1]
+      }
+    }
+  }
+}
+
+data "vcd_nsxt_tier0_router" "cluster_tier0_router" {
+  name            = var.cluster_nsxt_tier0_router_name
+  nsxt_manager_id = data.vcd_nsxt_manager.cse_nsxt_manager.id
+}
+
+resource "vcd_external_network_v2" "cluster_tier0" {
+  name = "cluster_tier0"
+
+  nsxt_network {
+    nsxt_manager_id      = data.vcd_nsxt_manager.cse_nsxt_manager.id
+    nsxt_tier0_router_id = data.vcd_nsxt_tier0_router.cluster_tier0_router.id
+  }
+
+  ip_scope {
+    gateway       = var.cluster_provider_gateway_gateway_ip
+    prefix_length = var.cluster_provider_gateway_gateway_prefix_length
+
+    dynamic "static_ip_pool" {
+      for_each = var.cluster_provider_gateway_static_ip_ranges
+      iterator = ip
+      content {
+        start_address = ip.value[0]
+        end_address   = ip.value[1]
+      }
+    }
+  }
+}
+
+# This Edge Gateway will consume automatically the available IPs from the Provider Gateway.
+resource "vcd_nsxt_edgegateway" "solutions_edgegateway" {
+  org      = vcd_org.solutions_organization.name
+  owner_id = vcd_org_vdc.solutions_vdc.id
+
+  name                = "solutions_edgegateway"
+  external_network_id = vcd_external_network_v2.solutions_tier0.id
+
+  # TODO: Change to automatic allocation!!!
```

Comment:
Change to automatic allocation :)

Reply:
#991 is merged :)

Reply:
While changing it I thought that I may leave as it is now, as it consumes all the available IPs this way and I don't need to add a variable to ask how many per subnet does the user want. (As this is an example, users could change it freely if they wish)